api/src/main/java/com/velocitypowered/api/proxy/server/PingOptions.java

```diff
@@ -0,0 +1,157 @@
+/*
+ * Copyright (C) 2018-2023 Velocity Contributors
+ *
+ * The Velocity API is licensed under the terms of the MIT License. For more details,
+ * reference the LICENSE file in the api top-level directory.
+ */
+
+package com.velocitypowered.api.proxy.server;
+
+import static com.google.common.base.Preconditions.checkArgument;
+
+import com.velocitypowered.api.network.ProtocolVersion;
+import java.time.Duration;
+import java.util.Objects;
+import java.util.concurrent.TimeUnit;
+import net.kyori.adventure.builder.AbstractBuilder;
+import org.jetbrains.annotations.NotNull;
+
+/**
+ * Contains the parameters used to ping a {@link RegisteredServer}.
+ * This class is immutable.
+ *
+ * @since 3.2.0
+ * @see RegisteredServer#ping(PingOptions)
+ */
+public final class PingOptions {
+  /**
+   * Default PingOptions.
+   * Uses a {@link ProtocolVersion#UNKNOWN} version and a timeout of 3000.
```

Comment:
Let's not document the specific default timeout, as it may be changed by the user.